graphql_compiler/compiler/emit_sql.py

```diff
@@ -112,3 +125,154 @@ def _get_output_columns(node, context):
+        node: SqlNode, the SqlNode the expression applies to.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Expression, SQLAlchemy expression.
+    """
+    if expression.operator in constants.UNSUPPORTED_OPERATOR_NAMES:
+        raise NotImplementedError(
+            u'Filter operation "{}" is not supported by the SQL backend.'.format(
+                expression.operator))
+    sql_operator = constants.SUPPORTED_OPERATORS[expression.operator]
+    left = _expression_to_sql(expression.left, node, context)
+    right = _expression_to_sql(expression.right, node, context)
+    if sql_operator.cardinality == constants.Cardinality.UNARY:
+        # ensure the operator is grabbed from the Column object
+        if not isinstance(left, Column) and isinstance(right, Column):
+            left, right = right, left
+        clause = getattr(left, sql_operator.name)(right)
+        return clause
+    elif sql_operator.cardinality == constants.Cardinality.BINARY:
+        clause = getattr(sql_expressions, sql_operator.name)(left, right)
+        return clause
+    elif sql_operator.cardinality == constants.Cardinality.LIST_VALUED:
+        if isinstance(left, BindParameter) and isinstance(right, Column):
+            left, right = right, left
+        if not isinstance(right, BindParameter):
+            raise AssertionError(
+                u'List valued operator expects column as left side of expression')
+        if not isinstance(left, Column):
+            raise AssertionError(
+                u'List valued operator expects bind parameter as right side of expression')
+        # ensure that SQLAlchemy treats the left bind parameter as list valued
+        right.expanding = True
+        clause = getattr(left, sql_operator.name)(right)
+        return clause
+    else:
+        raise AssertionError(u'Unknown operator cardinality {}'.format(sql_operator.cardinality))
+
+
+def _transform_literal_to_expression(expression, node, context):
+    """Transform a Literal compiler expression into it's SQLAlchemy expression representation.
+
+    Args:
+        expression: expression, Literal compiler expression.
+        node: SqlNode, the SqlNode the expression applies to.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Expression, SQLAlchemy expression.
+    """
+    return expression.value
+
+
+def _transform_variable_to_expression(expression, node, context):
+    """Transform a Variable compiler expression into it's SQLAlchemy expression representation.
+
+    Args:
+        expression: expression, Variable compiler expression.
+        node: SqlNode, the SqlNode the expression applies to.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Expression, SQLAlchemy expression.
+    """
+    variable_name = expression.variable_name
```

Comment:
`expression` is meant to be of type `Variable`, no? If so, then line 260 should be an assertion -- variable names are *required* to start with `$`.

Reply:
Good to know, this is an assertion now.